File: 2582-pass-the-pillow/2582-pass-the-pillow.rb
# @param {Integer} n
# @param {Integer} time
# @return {Integer}
def pass_the_pillow(n, time)
    # remove full cycles
    roundTrip = (n-1) * 2
    time %= roundTrip

    if time < n
        # left to right
        time + 1
    else
        # right to left
        time -= n
        n - time - 1
    end
end

File: 2582-pass-the-pillow/2582-pass-the-pillow.c
int passThePillow(int n, int time) {
    // remove full cycles
    int roundTrip = (n-1) * 2;
    time %= roundTrip;

    // left to right
    if(time < n) return time + 1;

    // right to left
    time -= n;
    return n - time - 1;
}